### PR TITLE
GPMSA Example: Brian Williams' linear verification

### PR DESCRIPTION
**Pull request by briadam**
@roystgnr @dmcdougall @brianw525 This starts to address #579 by implementing Brian's test problem in a QUESO example.  It runs and doesn't crash (issues a bunch of warnings from one particular spot though.)

In terms of your development workflow, I wasn't sure if you want this to be managed on a branch until the test is verified vs. merged to devel. Feel free to guide me as to how to work.

Add initial version of Bayes linear verification problem due to Brian
Williams

[1] Adams, B.M., Coleman, Kayla, Hooper, Russell W., Khuwaileh, Bassam
A., Lewis, Allison, Smith, Ralph C., Swiler, Laura P., Turinsky, Paul
J., Williams, Brian J., "User Guidelines and Best Practices for CASL
VUQ Analysis Using Dakota, Sandia Technical Report 2016-11614 (also
CASL-U-2016-1233-000), September 2016.

### CODE REVIEW

**Comment by briadam** [comment]
Spoke with @roystgnr who suggested if possible we make some or all of this a test, with XFAIL status if needed.  While the ultimate test @brianw525 has in mind likely won't be feasible for this (requires 100k or 250k samples) and is why I targeted examples/, we can make a scaled down version to use as at least a regression test.

**Comment by dmcdougall** [comment]
Do you want me to merge this example or wait for a test?

**Comment by briadam** [comment]
Yes, but I defer to you all as to what's appropriate in the project.  One of my goals prior to our next in-person meeting is to implement a few of Brian Williams' verification cases so we can actually iterate on working code in the meeting and run cases.  Since the cases have to run many samples, perhaps for now they belong as examples and not tests and that's just okay?  Also, since I'll be adding a few more cases similar to this, I am keenly interested in feedback on my implementation before I copy/paste and make other cases, but that's not required.

File: examples/gp/linear_verif/gpmsa_input.txt
Comment (dmcdougall) [inline examples/gp/linear_verif/gpmsa_input.txt]: Vectors need single quotes around them (`'0 1'`) because`dev` uses `GetPot` to parse input files, and not `boost`.  You can still use `boost` if you want, but the default is `GetPot`.

File: examples/gp/linear_verif/gpmsa_input.txt
Comment (dmcdougall) [inline examples/gp/linear_verif/gpmsa_input.txt]: Needs quotes.

File: examples/gp/linear_verif/gpmsa_input.txt
Comment (dmcdougall) [inline examples/gp/linear_verif/gpmsa_input.txt]: Needs quotes.

File: examples/gp/linear_verif/gpmsa_linear_verif.C
Comment (dmcdougall) [inline examples/gp/linear_verif/gpmsa_linear_verif.C]: Can you put the QUESO licence before this?  You can copy it from one of the other examples.

**Review by dmcdougall — CHANGES_REQUESTED** [review]
You can add your example to [this file](https://github.com/libqueso/queso/blob/dev/examples/run_all.sh) if you want it to be easily executed from our `run_all.sh` script.

Caveat other remarks, this looks fine to me.  Thanks!

**Comment by briadam** [comment]
Thanks for the helpful feedback.  Brian Williams updated this example and gave me several others, so I'm thinking I'll also reorganize the directories and do some cleanup, maybe even build some utilities that can be shared among the cases, before we merge...

**Comment by dmcdougall** [comment]
No worries.  Let me know when you're ready to pull the trigger.

**Review by dmcdougall — CHANGES_REQUESTED** [review]

File: examples/Makefile.am
Comment (roystgnr) [inline examples/Makefile.am]: This installs gpmsa_scalar.C, not gpmsa_linear_verif.C

**Pull request merged**
